src/hooks/use-and-require-context.ts

```diff
@@ -0,0 +1,16 @@
+import { type Context, useContext, useMemo } from "react";
+
+export function useAndRequireContext<T>(
+  context: Context<T | undefined>,
+  objectName = "hook",
+  providerName = "provider",
+): T {
+  const possiblyUndefinedContext = useContext(context);
+  return useMemo(() => {
+    if (possiblyUndefinedContext === undefined)
+      throw new Error(
+        `Attempt to use ${objectName} outside of ${providerName}`,
+      );
```

Comment:
Is this handled anywhere or will the app just crash?

Reply:
Added a TODO to tie it to the error boundary Carlos is adding